lmdeploy/pytorch/chat.py

```diff
@@ -70,19 +72,31 @@ def init_model(
         model = AutoModelForCausalLM.from_pretrained(model_path,
                                                      torch_dtype=torch.float16,
                                                      trust_remote_code=True)
-    model = model.cuda(local_rank)
 
     if not _is_deepspeed_available:
         warnings.warn('deepspeed is not installed, '
                       'use plain huggingface model.')
+
+        model = model.cuda(local_rank)
     else:
-        model = deepspeed.init_inference(
-            model=model,  # Transformers models
-            mp_size=world_size,  # Number of GPU
+        config = dict(
+            tensor_parallel=dict(tp_size=world_size),  # Number of GPU
             dtype=torch.float16,  # dtype of the weights (fp16)
             replace_with_kernel_inject=True,
             # replace the model with the kernel injector
-            max_out_tokens=2048,
+            max_out_tokens=max_new_tokens,
```

Comment:
I suggest not mixing `max_out_tokens` and `max_new_tokens`. The former in deepspeed is used to allocate memories for kv_cache & rotary_embedding (default to 1024 by deepspeed). I set it to  2048 because it is the default max content length for llama and internlm. And it seems (not 100% certain) deepspeed will automatically shrink this value based on available GPU memory. 

`max_new_tokens` is used to control the stopping criteria of hugging-face. It is somehow a dynamic variable in python and can be even changed during chat without restart the program.

If you want to save some memory, how about using a separate argument for this, like `max_tokens` and defaulting to 2048?

Reply:
Do you mean that max_new_tokens is the maximum value of text a round of Q&A can generate? How to control the total length of the context of multiple rounds of  Q&A?

Sorry, I didn't understand correctly that `max_out_tokens`  is used to allocate memories for kv_cache & rotary_embedding . I thought the two params had the same effect before. 

 I am not clear that If `max_new_tokens` > `max_out_tokens`  the maximum text that a round of Q&A can output depend on which params？